docs/how_to/pytorch_install/pytorch_install.md

```diff
@@ -1,377 +1,391 @@
 
-   ```dockerfile
-   docker run -it --device=/dev/kfd --device=/dev/dri --group-add video rocm/dev-ubuntu-20.04:latest
+   Pull selected base OS image (ubuntu 20.04 for example)
+
+   ```docker
+   docker pull ubuntu:20.04
    ```
 
-3. Install any dependencies needed for installing the wheels package.
+   Start a Docker container using the downloaded image
+
+   ```docker
+   docker run -it --device=/dev/kfd --device=/dev/dri --group-add video ubuntu:20.04
+   ```
+
+   Install ROCm using the directions in the [Installation section](#install_rocm_linux).
+
+   **Option 3:**
+
+   Install on bare metal. Check [OS compatibility](#supported_distributions) and install ROCm using the
+   directions in the [Installation section](#install_rocm_linux).
+
+2. Install the required dependencies for the wheels package.
 
    ```bash
    sudo apt update
-   sudo apt install libjpeg-dev python3-dev
+   sudo apt install libjpeg-dev python3-dev python3-pip
    pip3 install wheel setuptools
    ```
 
-4. Install torch, `torchvision`, and `torchaudio` as specified by the installation
-   matrix.
+3. Install `torch`, `torchvision`, and `torchaudio`, as specified in the [installation matrix](https://pytorch.org/get-started/locally/).
 
    :::{note}
-   ROCm 5.2 PyTorch wheel in the command below is shown for reference.
+   The following command uses the ROCm 5.6 PyTorch wheel. If you want a different version of ROCm,
+   modify the command accordingly.
    :::
 
    ```bash
-   pip3 install --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/rocm5.2/
+   pip3 install --pre torch torchvision torchaudio --index-url https://download.pytorch.org/whl/nightly/rocm5.6/
    ```
 
-#### Option 3: Install PyTorch Using PyTorch ROCm Base Docker Image
+4. (Optional) Use MIOpen kdb files with ROCm PyTorch wheels.
+
+   PyTorch uses [MIOpen](https://github.com/ROCmSoftwarePlatform/MIOpen) for machine learning
+   primitives, which are compiled into kernels at runtime. Runtime compilation causes a small warm-up
+   phase when starting PyTorch, and MIOpen kdb files contain precompiled kernels that can speed up
+   application warm-up phases. For more information, refer to the
+   {doc}`MIOpen installation page <miopen:install>`.
+
+   MIOpen kdb files can be used with ROCm PyTorch wheels. However, the kdb files need to be placed in
+   a specific location with respect to the PyTorch installation path. A helper script simplifies this task by
+   taking the ROCm version and GPU architecture as inputs. This works for Ubuntu and CentOS.
+
+   You can download the helper script here:
+   [install_kdb_files_for_pytorch_wheels.sh](https://raw.githubusercontent.com/wiki/ROCmSoftwarePlatform/pytorch/files/    install_kdb_files_for_pytorch_wheels.sh)
+
+   After installing ROCm PyTorch wheels, run the following code:
+
+   1. (Optional) `export GFX_ARCH=gfx90a`
```

Comment:
I don't like nested numbering with the same values as the outer set. Try a. b. and see what happens.

Reply:
Also, we could put all these steps in a code block, that might be better than numbering. Copy paste works better. Look at the install guide.

Reply:
Or just above in this file.